imperial_coldfront_plugin/views.py

```diff
@@ -631,3 +641,50 @@ def task_stat_view(request, group: str):
     return render(
         request, "imperial_coldfront_plugin/task_list.html", context={"tasks": task_qs}
     )
+
+
+def get_or_create_user(username: str) -> User:
+    """Get user from the database or creates one using data from Graph.
+
+    Args:
+        username: The username of the user to be retrieved or created.
+
+    Return:
+        The user, already existing or newly created.
+    """
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        client = get_graph_api_client()
+        user_data = client.user_profile(username)
+        user = User.objects.create(
+            username=user_data["username"],
+            first_name=user_data["first_name"],
+            last_name=user_data["last_name"],
+            email=user_data["email"],
+        )
+    return user
+
+
+def get_or_create_project(user: User) -> Project:
+    """Get project from the database or creates one.
+
+    Args:
+        user: The user object that will own a project.
+
+    Return:
+        The project, already existing or newly created.
+    """
+    project, project_created = Project.objects.get_or_create(
+        pi=user,
+        title=f"{user.get_full_name()}'s Research Group",
+        status=ProjectStatusChoice.objects.get_or_create(name="Active")[0],
+    )
```

Comment:
I'd just make a tiny tweak here to avoid creating a second group in the event that a project title gets edited or we decide to change the naming pattern later.
```suggestion
    project, project_created = Project.objects.get_or_create(
        pi=user,
        defaults=dict(
            title=f"{user.get_full_name()}'s Research Group",
            status=ProjectStatusChoice.objects.get_or_create(name="Active")[0],
        )
    )
```

Reply:
Well spotted!